src/lib/checkbox/checkbox.ts

```diff
@@ -18,66 +25,269 @@ declare global {
 }
 
 /**
- * The custom element class behind the `<forge-checkbox>` element.
- * 
  * @tag forge-checkbox
+ * 
+ * @summary Checkboxes select single values for submission in a form.
+ * 
+ * @description
+ * Use checkboxes to:
+ * - Select one or multiple items from a list.
+ * - Present a list containing sub-selections.
+ * - Turn an option on or off in desktop environment.
+ * 
+ * @property {boolean} checked - Whether the checkbox is checked.
+ * @property {boolean} defaultChecked - Whether the checkbox is checked by default.
+ * @property {boolean} indeterminate - Toggles the indeterminate state. This does not affect whether the checkbox is checked or its form submission.
+ * @property {string} value - The value of the checkbox when checked.
+ * @property {boolean} disabled - Controls if the checkbix is disabled.
+ * @property {boolean} required = Controls if the checkbox is required.
+ * @property {boolean} readonly = Controls if the checkbox is readonly.
+ * @property {boolean} dense - The density state.
+ * @property {SwitchLabelPosition} labelPosition - Whether the label appears before or after the checkbox.
+ * 
+ * @attribute {string} checked - Controls whether the checkbox is checked.
+ * @attribute {string} defaultChecked - Controls whether the checkbox is checked by default.
+ * @attribute {string} indeterminate - Controls the indeterminate state.
+ * @attribute {string} value - The value of the checkbox when checked.
+ * @attribute {string} disabled - Controls if the switch is disabled.
+ * @attribute {string} required - Controls if the switch is required.
+ * @attribute {string} readonly - Controls if the switch is readonly.
+ * @attribute {string} dense - Sets the density state.
+ * @attribute {string} label-position - Sets whether the label appears before or after the switch.
+ * 
+ * @method {(force?: boolean) => void} toggle - Toggles whether the checkbox is is checked or forces a checked state.
+ *  
+ * @event change {CustomEvent} - Dispatches when the switch's value changes.
```

Comment:
The type should come before the name here, and technically it's not a `CustomEvent` since we're redispatching, right?